src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Runtime/AvoidConstArrays.Fixer.cs

```diff
@@ -0,0 +1,170 @@
+    /// </summary>
+    [ExportCodeFixProvider(LanguageNames.CSharp, LanguageNames.VisualBasic), Shared]
+    public sealed class AvoidConstArraysFixer : CodeFixProvider
+    {
+        public sealed override ImmutableArray<string> FixableDiagnosticIds { get; } = ImmutableArray.Create(AvoidConstArraysAnalyzer.RuleId);
+
+        private static readonly ImmutableArray<string> s_collectionMemberEndings = ImmutableArray.Create("array", "collection", "enumerable", "list");
+
+        // See https://github.com/dotnet/roslyn/blob/main/docs/analyzers/FixAllProvider.md for more information on Fix All Providers
+        public sealed override FixAllProvider GetFixAllProvider() => WellKnownFixAllProviders.BatchFixer;
+
+        public override async Task RegisterCodeFixesAsync(CodeFixContext context)
+        {
+            Document document = context.Document;
+            SyntaxNode root = await document.GetSyntaxRootAsync(context.CancellationToken).ConfigureAwait(false);
+            SyntaxNode node = root.FindNode(context.Span);
+            SemanticModel model = await document.GetSemanticModelAsync(context.CancellationToken).ConfigureAwait(false);
+            DocumentEditor editor = await DocumentEditor.CreateAsync(document, context.CancellationToken).ConfigureAwait(false);
+            SyntaxGenerator generator = editor.Generator;
+
+            string title = MicrosoftNetCoreAnalyzersResources.AvoidConstArraysCodeFixTitle;
+            context.RegisterCodeFix(CodeAction.Create(
+                    title,
+                    async c => await ExtractConstArrayAsync(node, model, editor, generator, context.Diagnostics.First().Properties, c).ConfigureAwait(false),
+                    equivalenceKey: title),
+                context.Diagnostics);
+        }
+
+        private static async Task<Document> ExtractConstArrayAsync(SyntaxNode node, SemanticModel model, DocumentEditor editor,
+            SyntaxGenerator generator, ImmutableDictionary<string, string> properties, CancellationToken cancellationToken)
+        {
+            IArrayCreationOperation arrayArgument = GetArrayCreationOperation(node, model, cancellationToken, out bool isInvoked);
+            IEnumerable<ISymbol> members = model.GetEnclosingSymbol(node.SpanStart, cancellationToken).ContainingType.GetMembers();
+
+            // Get a valid member name for the extracted constant
+            string newMemberName = GetExtractedMemberName(
+                members.Where(x => x is IFieldSymbol).Select(x => x.Name),
+                properties["paramName"] ?? GetMemberNameFromType(arrayArgument));
+
+            // Get method containing the symbol that is being diagnosed.
+            IOperation? methodContext = arrayArgument.GetAncestor<IMethodBodyOperation>(OperationKind.MethodBody);
+            methodContext ??= arrayArgument.GetAncestor<IBlockOperation>(OperationKind.Block); // VB methods have a different structure than CS methods
```

Comment:
```suggestion
            IOperation? methodContext = arrayArgument.GetAncestor<IMethodBodyOperation>(OperationKind.MethodBody)
                ?? arrayArgument.GetAncestor<IBlockOperation>(OperationKind.Block); // VB methods have a different structure than CS methods
```

Reply:
For some reason that actually doesn't work, since `??` must be applied to operands of the same type. Using `??=` works fine since it just assigns if null, and `IBlockOperation` can be assigned to an `IOperation`.